Tests/scripts/destroy_instances.py

```diff
@@ -1,82 +1,144 @@
+                     f"exit code: {exit_code}")
 
-    except subprocess.CalledProcessError:
-        logging.exception(f'Failed to shutdown server {server_ip}')
+        return exit_code == 0
+    except SSHException:
+        logging.exception(f'Failed shutting down server {server_ip}')
+    return False
+
+
+def download_logs(ssh: SSHClient, server_ip: str, artifacts_dir: str, role: str) -> bool:
+
+    def progress(filename: bytes, size: int, sent: int, peer_name: tuple[str, int]):
+        logging.info(f"Downloading from {peer_name[0]}:{peer_name[1]} {filename!r} "
+                     f"progress: {float(sent) / float(size) * 100:.2f}% "
+                     f"({humanize.naturalsize(sent, binary=True, gnu=True)}/{humanize.naturalsize(size, binary=True, gnu=True)})")
+
+    try:
+        download_path = (Path(artifacts_dir) / f"server_{role}_{server_ip}.log").as_posix()
+        logging.info(f'Downloading server logs from server {server_ip} from:{SERVER_LOG_FILE_PATH} to {download_path}')
+        with SCPClient(ssh.get_transport(), progress4=progress) as scp:
+            scp.get(SERVER_LOG_FILE_PATH, download_path)
+        return True
+    except SCPException:
+        logging.exception(f'Failed downloading server logs from server {server_ip}')
+    return False
+
+
+def destroy_server(artifacts_dir: str, readable_role: str, role: str, server_ip: str, tests_path: Path,
+                   time_to_live: int) -> bool:
+    success = True
+    with SSHClient() as ssh:
+        try:
+            ssh.load_system_host_keys()
+            ssh.connect(server_ip, username=SSH_USER)
+            success &= chmod_logs(ssh, server_ip)
+            success &= download_logs(ssh, server_ip, artifacts_dir, role)
+
+            if time_to_live:
+                logging.info(f'Time to live was set to {time_to_live} minutes for server {server_ip}')
+                success &= shutdown(ssh, server_ip, time_to_live)
+            elif (tests_path / f'is_build_passed_{role}.txt').exists() and \
+                    (tests_path / f'is_post_update_passed_{role}.txt').exists():
+                success &= shutdown(ssh, server_ip)
+                logging.warning(f'Tests passed on {readable_role}, shutting down instance.')
+            else:
+                logging.warning(f'Tests for some integration failed on {readable_role}, keeping instance alive, '
+                                f'until server default TTL')
+        except SSHException:
+            logging.exception(f"Unable to SSH to server {server_ip}")
+            success = False
+    return success
 
 
 def main():
     install_logging('Destroy_instances.log')
     options = options_handler()
-    time_to_live = os.getenv('TIME_TO_LIVE')
+    time_to_live = int(os.getenv('TIME_TO_LIVE') or DEFAULT_TTL)
     tests_path = Path('./Tests')
+    start_time = datetime.utcnow()
+    logging.info(f"Starting destroy instances - environment from {options.env_file}, TTL: {time_to_live} seconds, "
```

Comment:
Seconds or minutes? Line 99 says minutes.
```suggestion
    logging.info(f"Starting destroy instances - environment from {options.env_file}, TTL: {time_to_live} minutes, "
```